docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    DictionaryKeyPolicy = JsonNamingPolicy.CamelCase
+};
+json = JsonSerializer.Serialize(weatherForecast, options);
+```
+
+Example object to serialize and JSON output:
+
+|Property |Value  |
+|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00|
+| TemperatureC| 25 |
+| Summary| Hot|
+| TemperatureRanges | Cold, 20<br>Hot, 40|
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot",
+  "TemperatureRanges": {
+    "cold": 20,
+    "hot": 40
+  }
+}
+```
+
+The camel case naming policy applies to serialization only.
+
+## Exclude properties
+
+This section explains how to exclude:
+
+* Individual properties
+* All read-only properties
+* All null-value properties 
+
+### Exclude individual properties
+
+Use the [[JsonIgnore]](xref:System.Text.Json.Serialization.JsonIgnoreAttribute) attribute:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+    [JsonIgnore]
+    public int WindSpeed { get; set; }
+}
+```
+
+Example JSON output:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot"
+}
+```
+
+### Exclude all read-only properties
+
+Set <xref:System.Text.Json.JsonSerializerOptions.IgnoreReadOnlyProperties> to true:
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    IgnoreReadOnlyProperties = true
+};
+json = JsonSerializer.Serialize(weatherForecast, options);
+```
+
+Example class to serialize and JSON output:
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+    public int WindSpeed { get; private set; }
+}
+```
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25,
+  "Summary": "Hot",
+}
+```
+
+This option applies only to serialization. During deserialization, read-only properties are ignored by default. A property is read-only if it contains a public getter but not a public setter.
+
+### Exclude all null value properties
+
+Set <xref:System.Text.Json.JsonSerializerOptions.IgnoreNullValues> to true:
+
+```csharp
+var options = new JsonSerializerOptions
+{
+    IgnoreNullValues = true
+};
+json = JsonSerializer.Serialize(weatherForecast, options);
+```
+
+Example object to serialize:
+
+|Property |Value  |
+|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00|
+| TemperatureC| 25 |
+| Summary| null|
+
+`Summary` is omitted from the JSON output:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 25
+}
+```
+
+This setting applies to serialization and deserialization. During deserialization, null values in the JSON are ignored only if they are valid. [Null values for non-nullable value types cause exceptions](https://github.com/dotnet/corefx/issues/40922).
```

Comment:
Can we add an example where a default value is not overriden by null on deserialize when IgnoreNullValues is set to true.